### PR TITLE
Initial support for deploying on OpenShift

### LINKED ISSUE #310: Initial support to deploy to OpenShift
### Description

Provide a new folder provision/openshift to be able to provision to OpenShift

Depending on the environment, some features like persistent storage or containers running as root might not be possible in the first version.

### Discussion

_No response_

### Motivation

Run load test also on OpenShift

### Details

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
It is possible to deploy Keycloak with legacy store and a PostgreSQL running on OpenShift.

Closes #310

### CODE REVIEW

**Comment by ahus1** [comment]
Hi @kami619 and @mhajas - this is an initial support to deploy to OpenShift. I'd be happy if one of you could review this. 

The existing CI is still green, but there is no CI for OpenShift in this repository. Functionality to enable monitoring in OpenShift and anything that requires persistent volumes hasn't been enabled yet.

File: provision/openshift/split.sh
Comment (mhajas) [inline provision/openshift/split.sh:1]: Do we need this file here as it is already in common?

File: provision/openshift/Taskfile.yaml
Comment (mhajas) [inline provision/openshift/Taskfile.yaml:26]:
```suggestion
      - common:split
```

**Review by mhajas — COMMENTED** [review]
Thank you @ahus1! This is a great initial work for Openshift support, now we can work on this together and divide the work for each missing part. 

I added a few comments into the code.

File: provision/openshift/Taskfile.yaml
Comment (ahus1) [reply provision/openshift/Taskfile.yaml]: applied

File: provision/openshift/Taskfile.yaml
Comment (ahus1) [reply provision/openshift/Taskfile.yaml:51]: applied

File: provision/openshift/Taskfile.yaml
Comment (ahus1) [reply provision/openshift/Taskfile.yaml:16]: applied

File: doc/kubernetes/modules/ROOT/pages/openshift.adoc
Comment (ahus1) [reply doc/kubernetes/modules/ROOT/pages/openshift.adoc]: applied

**Comment by ahus1** [comment]
@mhajas - thank you for that thorough review and the good input. I've accepted all your changes, and the build is green again. I also did a local test and it looks good to me. 

Please give it another review. Thanks!

File: doc/kubernetes/modules/ROOT/pages/openshift.adoc
Comment (kami619) [inline doc/kubernetes/modules/ROOT/pages/openshift.adoc:12]: do we want to point to some docs to establishing connection to a remote openshift cluster, such as login vs kubeconfig and is there any preferred way of doing that ?

Comment (ahus1) [reply doc/kubernetes/modules/ROOT/pages/openshift.adoc:12]: @kami619 - it would be good to have docs and maybe pointers to additional docs for that. Please create a new issue for this so we can tackle this after this PR has been merged.

Comment (kami619) [reply doc/kubernetes/modules/ROOT/pages/openshift.adoc:12]: wfm, I will create the issue. thanks Alex.

File: provision/openshift/isup.sh
Comment (kami619) [inline provision/openshift/isup.sh:19]: shouldn't this be `if [ "${SERVICE}" == "keycloak.${KC_HOSTNAME_SUFFIX}" ]` instead of `if [ "${SERVICE}" == "keycloak-${KC_HOSTNAME_SUFFIX}" ]`. Basically `.` instead of a `-` as defined in line 12 ?

**Comment by kami619** [comment]
@ahus1 apart from the one change I requested above, rest all looks great and it is tested with different storage combinations with the postgresql database and they seem to be working without issue. Good to merge this once the last piece is sorted.

**Review by kami619 — APPROVED** [review]
lgtm @ahus1 apart from the one change I requested above, rest all looks great and it is tested with different storage combinations with the postgresql database and they seem to be working without issue. Good to merge this once the last piece is sorted.

File: provision/openshift/isup.sh
Comment (ahus1) [reply provision/openshift/isup.sh]: Thanks - fixed that in the next commit. This should prevent the symptoms you've seen earlier when "isup" was showing that a storage change was complete when it actually wasn't.

**Pull request merged**
